File: src/main/java/com/partymakers/shareparty/data/persistence/friends/entity/FriendMapping.java
package com.partymakers.shareparty.data.persistence.friends.entity;

import com.partymakers.shareparty.data.persistence.helper.DomainMapper;
import com.partymakers.shareparty.data.persistence.helper.PersistenceMapper;
import com.partymakers.shareparty.domain.friends.entity.Friend;

public class FriendMapping implements DomainMapper<Friend, FriendEntity>, PersistenceMapper<FriendEntity, Friend> {

    @Override
    public FriendEntity toPersistence(Friend domainEntity) {
        return new FriendEntity(domainEntity.getName(), domainEntity.getNickName(), domainEntity.getEMail());
    }

    @Override
    public Friend toDomain(FriendEntity persistenceEntity) {
        return new Friend(persistenceEntity.getName(), persistenceEntity.getNickName(), persistenceEntity.getEMail());
    }
}

File: src/main/kotlin/com/partymakers/shareparty/data/persistence/friends/entity/FriendMapping.kt
package com.partymakers.shareparty.data.persistence.friends.entity

import com.partymakers.shareparty.data.persistence.helper.DomainMapper
import com.partymakers.shareparty.data.persistence.helper.PersistenceMapper
import com.partymakers.shareparty.domain.friends.entity.Friend

class FriendMapping : DomainMapper<Friend, FriendEntity>, PersistenceMapper<FriendEntity, Friend> {

    override fun toPersistence(domainEntity: Friend): FriendEntity =
        FriendEntity(domainEntity.name, domainEntity.nickName, domainEntity.eMail)

    override fun toDomain(persistenceEntity: FriendEntity): Friend =
        Friend(persistenceEntity.name.orEmpty(),
               persistenceEntity.nickName.orEmpty(),
               persistenceEntity.eMail.orEmpty())
}